crates/rpc/rpc/src/debug.rs

```diff
@@ -644,14 +647,43 @@ where
                     })
                     .map_err(|err| EthApiError::Internal(err.into()))?;
 
-                let ExecutionWitnessRecord { hashed_state, codes, keys } = witness_record;
+                let ExecutionWitnessRecord { hashed_state, codes, keys, lowest_block_number } =
+                    witness_record;
 
                 let state = state_provider
                     .witness(Default::default(), hashed_state)
                     .map_err(EthApiError::from)?;
-                Ok(ExecutionWitness { state, codes, keys })
+                Ok((
+                    ExecutionWitness { state, codes, keys, headers: Vec::new() },
+                    lowest_block_number,
+                ))
             })
-            .await
+            .await?;
+
+        let smallest = match lowest_block_number {
+            Some(smallest) => smallest,
+            None => {
+                // Return only the parent header, if there were no calls to the
+                // BLOCKHASH opcode.
+                block_number.saturating_sub(1)
+            }
+        };
+
+        let range = smallest..block_number;
```

Comment:
If the block_number is 5 for example, then smallest = 4 and range will be 4..5 (not including 5, so just 4)